nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/util/FlowDifferenceFilters.java

```diff
@@ -113,4 +117,145 @@ public static boolean isIgnorableVersionedFlowCoordinateChange(final FlowDiffere
 
         return false;
     }
+
+
+    public static boolean isNewPropertyWithDefaultValue(final FlowDifference fd, final FlowManager flowManager) {
+        if (fd.getDifferenceType() != DifferenceType.PROPERTY_ADDED) {
+            return false;
+        }
+
+        final VersionedComponent componentB = fd.getComponentB();
+
+        if (componentB instanceof InstantiatedVersionedProcessor) {
+            final InstantiatedVersionedProcessor instantiatedProcessor = (InstantiatedVersionedProcessor) componentB;
+            final ProcessorNode processorNode = flowManager.getProcessorNode(instantiatedProcessor.getInstanceId());
+            return isNewPropertyWithDefaultValue(fd, processorNode);
+        } else if (componentB instanceof InstantiatedVersionedControllerService) {
+            final InstantiatedVersionedControllerService instantiatedControllerService = (InstantiatedVersionedControllerService) componentB;
+            final ControllerServiceNode controllerService = flowManager.getControllerServiceNode(instantiatedControllerService.getInstanceId());
+            return isNewPropertyWithDefaultValue(fd, controllerService);
+        }
+
+        return false;
+    }
+
+    private static boolean isNewPropertyWithDefaultValue(final FlowDifference fd, final ProcessorNode processorNode) {
```

Comment:
This method appears to be overloaded below. However, the parameter that differs shares a common ancestor interface. I _think_ we can update the args to use a `ComponentNode` and remove the duplicated code.

Reply:
D'oh, good point! Will push a new commit that takes advantage of this.